persistence/src/main/scala/com/lightbend/lagom/internal/persistence/ReadSideActor.scala

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.persistence
+
+import akka.actor.Status.Failure
+import akka.{ Done, NotUsed }
+import akka.actor.{ Actor, ActorLogging, ActorRef, Props, Status }
+import akka.stream.javadsl.Source
+import akka.stream.scaladsl.{ Keep, Sink }
+import akka.stream.{ KillSwitch, KillSwitches, Materializer }
+import akka.util.Timeout
+import com.lightbend.lagom.internal.persistence.GlobalPrepareReadSideActor.Prepare
+import com.lightbend.lagom.internal.persistence.cluster.ClusterDistribution.EnsureActive
+import com.lightbend.lagom.javadsl.persistence._
+
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.duration.FiniteDuration
+
+private[lagom] object ReadSideActor {
+
+  def props[Event <: AggregateEvent[Event]](
+    processor:            () => ReadSideProcessor[Event],
+    eventStreamFactory:   (AggregateEventTag[Event], Offset) => Source[akka.japi.Pair[Event, Offset], NotUsed],
+    clazz:                Class[Event],
+    globalPrepare:        ActorRef,
+    globalPrepareTimeout: FiniteDuration
+  )(implicit mat: Materializer) = {
+    Props(classOf[ReadSideActor[Event]], processor, eventStreamFactory, clazz, globalPrepare, globalPrepareTimeout, mat)
+  }
+
+  /**
+   * Start processing from the given offset
+   */
+  case class Start(offset: Offset)
+}
+
+/**
+ * Read side actor
+ */
+private[lagom] class ReadSideActor[Event <: AggregateEvent[Event]](
+  processorFactory:     () => ReadSideProcessor[Event],
+  eventStreamFactory:   (AggregateEventTag[Event], Offset) => Source[akka.japi.Pair[Event, Offset], NotUsed],
+  clazz:                Class[Event],
+  globalPrepare:        ActorRef,
+  globalPrepareTimeout: FiniteDuration
+)(implicit mat: Materializer) extends Actor with ActorLogging {
+  import ReadSideActor._
+  import akka.pattern.pipe
+  import akka.pattern.ask
+  import context.dispatcher
+
+  private var shutdown: Option[KillSwitch] = None
+
+  override def postStop: Unit = {
+    shutdown.foreach(_.shutdown())
+  }
+
+  def receive = {
+    case EnsureActive(tagName) =>
+
+      val tag = AggregateEventTag.of(clazz, tagName)
+
+      implicit val timeout = Timeout(globalPrepareTimeout)
+
+      globalPrepare ? Prepare pipeTo self
+      context become preparing(tag)
+  }
+
+  def preparing(tag: AggregateEventTag[Event]): Receive = {
+
+    case Done =>
+      val handler = processorFactory().buildHandler()
+      handler.prepare(tag).toScala.map(Start(_)) pipeTo self
+      context become active(handler, tag)
+
+    case Status.Failure(e) =>
+      throw e
+
+  }
```

Comment:
@patriknw ReadSideActor now has an intermediate preparing state while it waits for the global prepare cluster singleton to prepare.